File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):
    
    array = []
    for i in range(1, number+1):
        count = 0
        for j in range(1, int(i**0.5)+1):
            if i % j == 0:
                count += 1
                if j**2 != i:
                    count += 1
            if count > limit:
                count = power
                break
        array.append(count)
    
    return sum(array)

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.java
class Solution {
    public int solution(int number, int limit, int power) {
        
        int[] array = new int[number];
        int result = 0;
        for(int i=1; i<=number; i++) {
            int answer = 1;
            for(int j=1; j<=i/2; j++) {
                if(i % j == 0) {
                    answer++;
                }
            }
        
            if (answer <= limit) {
                result += answer;
            } else {
                result += power;
            }
        }
        
        return result;
    }
}